plutus-ledger-api/src/Plutus/V1/Ledger/EvaluationContext.hs

```diff
@@ -53,10 +54,19 @@ inlining).
 -- | Build the 'EvaluationContext'.
 --
 -- The input is a `Map` of strings to cost integer values (aka `Plutus.CostModelParams`, `Alonzo.CostModel`)
-mkEvaluationContext :: Plutus.CostModelParams
-                    -> Maybe EvaluationContext
-mkEvaluationContext newCMP =
-    EvaluationContext . inline Plutus.mkMachineParameters <$> Plutus.applyCostModelParams Plutus.defaultCekCostModel newCMP
+mkEvaluationContextUnliftingImmediate :: Plutus.CostModelParams -> Maybe EvaluationContext
```

Comment:
This distinction isn't the ledger team's problem, rather we will make the decision based on the protocol version and the language version. So this function would need to take a protocol version - I'll check that that's okay with the ledger team.

Reply:
For now, I think just put this back the way it was and pass `defaultUnliftingMode` and we (i.e. @nikos ;) ) will sort it out afterwards.

Reply:
In fact, I realised from asking Jared about it that probably what we should do for now is this:
- Put both the deferred _and_ the immediate unlifting versions into the `EvaluationContext`
- Pull out the one that we need in `evaluateScript` based on the protocol version

That way we avoid requiring a protocol version in `mkEvaluationContext` (although we may want one eventually anyway). Doable here because there are only two options!

Reply:
Great, makes sense to me.

Reply:
After reading these comments, now I understand why we need both unlifting modes: Are we planning to switch to another unlifting mode in new language version?